web/templates/stream_settings/stream_types.hbs

```diff
@@ -74,7 +74,7 @@
         {{#if (or is_owner is_stream_edit)}}
             <div>
                 <div class="input-group inline-block message-retention-setting-group time-limit-setting">
-                    <label class="dropdown-title">{{t "Message retention period" }}
+                    <label class="dropdown-title" for="id_message_retention_days">{{t "Message retention period" }}
```

Comment:
We're currently leaving this interaction to the browser. Clicking on the labels for `select` elements also doesn't work for me on Firefox. While on Chrome it would highlight the box:

![image](https://github.com/user-attachments/assets/0cf7cc26-a5c5-429b-9c7b-72670fc2c10d)

I think it might be best if we start a discussion about how we want to handle this and leave that as a follow up.

Reply:
I meant the input box shown for entering the custom input. I guess that would just need adding the correct for attribute and we would be fine with the default browser behavior.